### PR TITLE
Throws a validation exception now if attempting to cast an object to string

### PR DESCRIPTION
**Pull request by ronnied**
# Overview

PHP 7.4 was failing unit tests (PHPUnit v4.8.36).

Please preserve this line to notify @courtney-miles (lead of this repository)

### CODE REVIEW

**Comment by courtney-miles** [comment]
I can see this is due to a undocumented BC breakage with PHP 7.4 where previously it would throw an instance of `\Exception` and now it will throw an instance of `\Error`.

I believe that the correct solution is that it should expect to catch `\Throwable` rather than `\Exception`.  However, this would break compatibility with PHP 5.

Let me see if I can get acceptance for dropping support for PHP 5.

File: src/Fields/StringField.php [hunk 1]
```diff
@@ -19,11 +21,14 @@ public function inferProperties($val, $lenient = false)
...
+     * @throws FieldValidationException;
      */
     protected function validateCastValue($val)
     {
         try {
+            if (is_object($val)) {
+                throw $this->getValidationException('Could not cast object to string', $val);
+            }
             $val = (string) $val;
         } catch (\Exception $e) {
```
Comment (courtney-miles) [inline src/Fields/StringField.php]: The latest set of changes in `frictionlessdata:master` drop support for PHP 5.6.

With support for 5.6 dropped, I believe a better correction for PHP 7.4 will the following.
```suggestion
        try {
            $val = (string) $val;
        } catch (\Throwable $e) {
```

**Review by courtney-miles — CHANGES_REQUESTED** [review]
Hey @ronnied ,

Where this PR is for compatibility with PHP 7.4, can you also modify `.travis.yml` to add PHP 7.4 to be tested, and modify `composer.json` to remove the upper cap on PHP versions? (I.e. change the version constraint from `>=7.1.0 <7.4.0` to `>=7.1.0`.)

File: src/Fields/StringField.php (hunk 1 above)
Comment (ronnied) [reply src/Fields/StringField.php]: Done.

**Comment by courtney-miles** [comment]
Hey @ronnied ,

If you can allow me to contribute to this PR (and #43 ) I am happy to make the final corrections to get them to pass.

Thanks

**Pull request merged**